ee/api/chalicelib/core/tenants.py

```diff
@@ -78,7 +78,14 @@ def edit_tenant(tenant_id, changes):
         return helper.dict_to_camel_case(cur.fetchone())
 
 
-def tenants_exists(use_pool=True):
+def tenants_exists_sync(use_pool=True):
     with pg_client.PostgresClient(use_pool=use_pool) as cur:
-        cur.execute(f"SELECT EXISTS(SELECT 1 FROM public.tenants)")
+        cur.execute("SELECT EXISTS(SELECT 1 FROM public.tenants)")
         return cur.fetchone()["exists"]
+
+
+async def tenants_exists(use_pool=True):
+    async with application.get().database as cnx:
+        async with cnx.transaction() as txn:
+            row = await txn.execute(f"SELECT EXISTS(SELECT 1 FROM public.tenants)")
+            return await row.fetchone()["exists"]
```

Comment:
```suggestion
        async with cnx.transaction():
            await cnx.execute(f"SELECT EXISTS(SELECT 1 FROM public.tenants)")
            return await cnx.fetchone()["exists"]
```

Reply:
I mis-remember the psycopg v2 vs. v3 API.

For the record, the expected psycopg3's async api looks like the following pseudo code:

```python
    async with orpy.get().database.connection() as cnx:
         async with cnx.transaction():
             row = await cnx.execute("SELECT EXISTS(SELECT 1 FROM public.tenants)")
             row = await row.fetchone()
             return row["exists"]
```

Minding the following:

- Where `orly.get().database` is the postgresql connection pooler.
- Wrap explicit transaction with `async with cnx.transaction(): foobar()`
- Most of the time the transaction object is not used;
- Do execute await operation against `cnx`;
- `await cnx.execute` returns a cursor object;
- Do the `await cursor.fetchqux...` calls against the object return by a call to execute.